### PR TITLE
Set new date in the future if year is inferred

### LINKED ISSUE #697: default year should be such that event is in the future
So far the default year is always the current year. Wouldn't it be more handy if khal understands by default that I mean an event in the future?   If, in December 2017, I add a new event for the 20.04. , it should understand that I mean 20.04.2018, not 2017.

### PR DESCRIPTION
**Pull request by pfertyk**
Fixes #697.

Note: I still need to add tests for this.

### CODE REVIEW

**Comment by pfertyk** [comment]
Existing tests should be fixed, new test was added.

File: tests/utils_test.py [hunk 1]
```diff
@@ -390,11 +390,16 @@ def test_construct_daynames():
         'DTEND;TZID=Europe/Berlin;VALUE=DATE-TIME:20131026T000000')),
 
     # start and end date same, explicit end date (but no year) given
-    # XXX FIXME: if no explicit year is given for the end, this_year is used
     ('25.10.2013 18:00 26.10. 20:00 Äwesöme Event',
      _create_vevent(
         'DTSTART;TZID=Europe/Berlin;VALUE=DATE-TIME:20131025T180000',
-        'DTEND;TZID=Europe/Berlin;VALUE=DATE-TIME:20131026T200000')),
+        'DTEND;TZID=Europe/Berlin;VALUE=DATE-TIME:20141026T200000')),
```
Comment (geier) [inline tests/utils_test.py]: I don't think this should generate an event lasting for more than a year (and the FIXME is outdated any way).

**Comment by geier** [comment]
It probably should work like this:
* if only a time is given, the next time that time occurs is meant
* if a date without a year is given, the next time that date is occurs is meant *unless* it follows after a date including a year, than the next time the second date occurs after the first date is meant

we probably need some more tests for guessdatetimefstsr as well.

**Comment by geier** [comment]
I have merged these changes into #715 (and hopefully soon into master). Thanks again!

**Comment by pfertyk** [comment]
I'm sorry that I didn't finish the task, I was very busy recently :disappointed:. I'd like to implement the functionality you described, but I can't say exactly when it will be possible.

**Comment by geier** [comment]
No need, perhaps some more cleaning up to do, but I think we are good.
I applied a minor patch to your PR and added some tests and merged it
[0].  Thanks for taking care of this!


[0] https://github.com/pimutils/khal/pull/715/commits

Quoting pfertyk (2017-10-02 15:43:03)

**Pull request closed without merging**